struct.c

```diff
@@ -510,65 +509,123 @@ rb_struct_define_under(VALUE outer, const char *name, ...)
+ *    Foo.superclass                      # => Struct
+ *
+ *  Without string argument +class_name+,
+ *  returns a new anonymous subclass of +Struct+:
+ *
+ *    Struct.new(:foo, :bar).name # => nil
+ *
+ *  <b>Member Names</b>
+ *
+ *  With symbol arguments +member_names+,
+ *  returns a new subclass of +Struct+ with the given members:
+ *
+ *    Struct.new(:foo, :bar).members        # => [:foo, :bar]
+ *    Struct.new('Foo', :foo, :bar).members # => [:foo, :bar]
+ *
+ *  The new subclass has instance methods corresponding to +member_names+:
+ *
+ *    Foo = Struct.new('Foo', :foo, :bar)
+ *    Foo.instance_methods(false) # => [:foo, :bar, :foo=, :bar=]
+ *    f = Foo.new                 # => #<struct Struct::Foo foo=nil, bar=nil>
+ *    f.foo                       # => nil
+ *    f.foo = 0                   # => 0
+ *    f.bar                       # => nil
+ *    f.bar = 1                   # => 1
+ *    f                           # => #<struct Struct::Foo foo=0, bar=1>
+ *
+ *  <b><tt>Singleton Methods</tt></b>
+ *
+ *  The new subclass also has these singleton methods:
+ *
+ *    Foo = Struct.new('Foo', :foo, :bar)
+ *    Foo.singleton_methods # => [:[], :new, :inspect, :members]
+ *
+ *  - \Method <tt>:[]</tt> creates an instance of the subclass:
+ *
+ *      Foo[]        # => #<struct Struct::Foo foo=nil, bar=nil>
+ *      Foo[0]       # => #<struct Struct::Foo foo=0, bar=nil>
+ *      Foo[0, 1]    # => #<struct Struct::Foo foo=0, bar=1>
+ *      Foo[0, 1, 2] # Raises ArgumentError: struct size differs
+ *
+ *  - \Method <tt>:new </tt> also creates an instance of the subclass:
+ *
+ *      Foo.new          # => #<struct Struct::Foo foo=nil, bar=nil>
+ *      Foo.new(0)       # => #<struct Struct::Foo foo=0, bar=nil>
+ *      Foo.new(0, 1)    # => #<struct Struct::Foo foo=0, bar=1>
+ *      Foo.new(0, 1, 2) # Raises ArgumentError: struct size differs
+ *
+ *  - \Method <tt>:inspect</tt> returns a string representation of the subclass:
+ *
+ *      Foo.inspect
+ *      # => "Struct::Foo"
+ *
+ *  - \Method <tt>:members</tt> returns an array of the member names:
+ *
+ *      Foo.members # => [:foo, :bar]
+ *
+ *  <b>Keyword Argument</b>
+ *
+ *  By default, the arguments for initializing an instance of the new subclass
+ *  are ordinary arguments (not keyword arguments).
+ *  With optional keyword argument <tt>keyword_init: true</tt>,
+ *  the new subclass is initialized with keyword arguments:
+ *
+ *    # Without keyword_init: true.
+ *    Foo = Struct.new('Foo', :foo, :bar)
+ *    Foo                     # => Struct::Foo
+ *    Foo.new(0, 1)           # => #<struct Struct::Foo foo=0, bar=1>
+ *    # With keyword_init: true.
+ *    Bar = Struct.new(:foo, :bar, keyword_init: true)
+ *    Bar # =>                # => Bar(keyword_init: true)
+ *    Bar.new(bar: 1, foo: 0) # => #<struct Bar foo=0, bar=1>
+ *
+ *  <b>Block</b>
+ *
+ *  With a block given, all is as above, except that the block
```

Comment:
`With a block given, the block is yielded the created subclass:`.

I would move this section above the Member names section, since this describes behavior of the `Struct.new` method, not the behavior of the subclasses of Struct created by `Struct.new` (`Struct.new.new` ignores blocks).

Reply:
Changed to:  "With a block given, the created subclass is yielded to the block:"